Fix upgrade-versions command

olavphibm opened the pull request.

Commit:
Fix upgrade-versions command

When migrating from the old yaml sources format to the new one, the
commit that modified the upgrade_versions module
(1161796bbb0408eb981fbebf25c79bdd107b3f65) was incomplete, preventing
the spec release bumps.

The spec release bump and git commit logging were broken since commit
0839c75a5c7f528e7ca090d0542b04e843f20902. Only version updates were
working, but because the release bump was not even happening, the issue
was not observed.

Commit:
Do not include the previous commit in the spec changelog

The last commit of the previous spec release bump was being repeated as
if it were modified by the current bump.

alanoe commented on tools/upgrade_versions.py:
```diff
@@ -114,13 +114,17 @@ def update(self, user_name, user_email):
-        if self.pkg.commit_id:
+        old_commit_id = self.pkg.sources[0]["git"]["commit_id"]
+        if old_commit_id:
+            new_commit_id = pkg.sources[0]["git"]["commit_id"]
```
I'd prefer to put pkg.sources[0]["git"] in a new_source variable instead of duplicating it here and below, but up to you

olavphibm replied:
Done.

alanoe commented on tools/upgrade_versions.py:
```diff
@@ -114,13 +114,17 @@ def update(self, user_name, user_email):
 
     def _bump_release(self, pkg, log=None, user_name=None, user_email=None):
```
Does it make sense to accept a list here? I'd accept a str instead and create the log list with the str as the list first element. This would remove the need of making changelog variable above a list, which does not make sense IMHO

olavphibm replied:
Using a list isn't useful for our current usage, I'm applying your suggestion and renaming the variable to something better.

Commit:
Rename change log related variables

This makes it easier to undestand their purpose.
Some of those variables are being used incorrectly and the
update-versions command is broken, next commit fixes it.
